src/server/game/Auction/AsyncAuctionOperation.cpp

```diff
@@ -0,0 +1,502 @@
+/*
+ * This file is part of the AzerothCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by the
+ * Free Software Foundation; either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AsyncAuctionOperation.h"
+#include "AuctionHouseMgr.h"
+#include "AuctionHousePackets.h"
+#include "Creature.h"
+#include "DatabaseEnv.h"
+#include "GameTime.h"
+#include "ObjectAccessor.h"
+#include "Player.h"
+#include "ScriptMgr.h"
+
+Player* AsyncAuctionOperation::GetPlayer() const
+{
+    return ObjectAccessor::FindPlayer(_playerGuid);
+}
+
+void SellItemTask::Execute()
+{
+    auto player{ GetPlayer() };
+    if (!player)
+        return;
+
+    auto session{ player->GetSession() };
+
+    if (_packet->ItemsCount > MAX_AUCTION_ITEMS)
+    {
+        session->SendAuctionCommandResult(nullptr, AUCTION_SELL_ITEM, ERR_AUCTION_DATABASE_ERROR);
+        return;
+    }
+
+    for (uint32 i = 0; i < _packet->ItemsCount; ++i)
+    {
+        auto const& [itemGuid, itemCount] = _packet->Items[i];
+        if (!itemGuid || !itemCount || itemCount > 1000)
+        {
+            session->SendAuctionCommandResult(nullptr, AUCTION_SELL_ITEM, ERR_AUCTION_ITEM_NOT_FOUND);
+            return;
+        }
+    }
+
+    if (!_packet->Bid || !_packet->ExpireTime)
+    {
+        session->SendAuctionCommandResult(nullptr, AUCTION_SELL_ITEM, ERR_AUCTION_DATABASE_ERROR);
+        return;
+    }
+
+    if (_packet->Bid > MAX_MONEY_AMOUNT || _packet->Buyout > MAX_MONEY_AMOUNT)
+    {
+        LOG_ERROR("network", "WORLD: HandleAuctionSellItem - Player {} ({}) attempted to sell item with higher price than max gold amount.", player->GetName(), player->GetGUID().ToString());
```

Comment:
Should this really be classed as network? Why dont we combine all auctionhouse related messages into an auctionhouse logger?